enterprise/admin/views.py

```diff
@@ -291,6 +291,13 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
 
         return all_processable_emails
 
+    # TODO: this method is HUGE (> 100 lines) and complicated. Refactor it.
+    # Potential improvements:
+    # 1. Restructure flow so that notifications are sent after users and pending users are enrolled
+    #    FIXME: `send_email_notification_message` for pending users are executed OUTSIDE of ConditionalEmailConnection
```

Comment:
@mattdrayer I've logged a follow-up ticket in Opencraft board - please discuss with Braden if it makes sense to have it exposed as edX ticket as well.

Also, just to clarify, if I need something refactored/fixed or tech debt repaid I usually just do that and only resort to TODOs and FIXMEs when I'm *not* planning to address it in scope of current work, because the improvement suggested does not fit in scope ("does not fit in scope" basically means it would increase the amount of work to be done at least 75-100%).

This task as a whole was supposed to be a small one - just wrap Course Catalog API return values so that they are always consistent with what the rest of  edx-enterprise expects. But there was that problem with dependencies management I had to fix. In short, I've already exhausted "drive-by fixes and improvements" budget for this task :)